sw/device/silicon_creator/lib/boot_data.c

```diff
@@ -67,31 +67,90 @@ static rom_error_t boot_data_digest_compute(const void *boot_data,
                 "`digest` must be the first field of `boot_data_t`.");
 
   hmac_sha256_init();
-  RETURN_IF_ERROR(hmac_sha256_update(
-      (const char *)boot_data + kDigestRegionOffset, kDigestRegionSize));
-  RETURN_IF_ERROR(hmac_sha256_final(digest));
-  return kErrorOk;
+  hmac_sha256_update((const char *)boot_data + kDigestRegionOffset,
+                     kDigestRegionSize);
+  hmac_sha256_final(digest);
 }
 
+/**
+ * Shares for producing the `is_valid` value in `boot_data_digest_check()`.
+ * First 7 shares are generated using the `sparse-fsm-encode` script while the
+ * last share is `kHardenedBoolTrue ^ kHardenedBoolFalse ^ kDigestShares[0] ^
+ * ... ^ kDigestShares[7]` so that xor'ing all shares with the initial value of
+ * `is_valid`, i.e. `kHardenedBoolFalse`, produces `kHardenedBoolTrue`.
+ *
+ * Encoding generated with
+ * $ ./util/design/sparse-fsm-encode.py -d 6 -m 7 -n 32 \
+ *     -s 266012770 --language=c
+ *
+ * Hamming distance histogram:
```

Comment:
I think it is ok to remove the histogram, or just keep the relevant lines.

Reply:
Done, removed the histogram, just kept min/max HW/HD.